server/search/facets.go

```diff
@@ -0,0 +1,202 @@
+// Copyright 2022 Tigris Data, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package search
+
+import (
+	"github.com/rs/zerolog/log"
+	"github.com/tigrisdata/tigris/lib/container"
+	tsApi "github.com/typesense/typesense-go/typesense/api"
+	"github.com/pkg/errors"
+	api "github.com/tigrisdata/tigris/api/server/v1"
+)
+
+// SortedFacets is a Temporary workaround to merge facet values when aggregating results from
+// multi-search queries resulting from OR filters
+// this is not very efficient as of now
+type SortedFacets struct {
+	counts     map[string]*container.PriorityQueue[FacetCount]
+	facetAttrs map[string]*FacetAttrs
+	sorted     bool
+}
+
+func NewSortedFacets() *SortedFacets {
```

Comment:
Not for now but should we introduce an abstraction so that we don't use a priority queue if OR queries are not issued by the user?